src/hooks/useAddItem.js

```diff
@@ -0,0 +1,89 @@
+import { useState } from 'react';
+import { db } from '../lib/firebase';
+import { addDoc, collection } from 'firebase/firestore';
+import useFetchItems from './useFetchItems';
+
+const useAddItem = (reference) => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [successMessage, setSuccessMessage] = useState('');
+  const [error, setError] = useState(null);
+  const [duplicateItemMessage, setDuplicateItemMessage] = useState('');
+  const { data: items } = useFetchItems();
+
+  const addItem = async (newItem, timeframe, token, lastPurchased = null) => {
+    setIsLoading(true);
+    try {
+      const ListRef = collection(db, 'Lists');
+      let addedDocument;
+      checkDuplication(newItem)
+        ? showErrorMessage()
+        : (addedDocument = await addDoc(ListRef, {
+            itemName: newItem,
+            timeframe: parseInt(timeframe),
+            lastPurchased,
+            token,
+          }));
+      if (addedDocument) {
+        setIsLoading(false);
+        showSuccesMessage();
+        setError(false);
+      }
+    } catch (e) {
+      setError(true);
+      setSuccessMessage(null);
+    }
+  };
+
+  //-------------------------------------------------------------------------------------
+  // HELPER FUNCTIONS
+
+  //Check for duplication:
+  //get the existing items list from firebase
+  //loop through the existing items list to check if there is a match with current item
+  //set itemExists to true if duplication and return it
+
+  const checkDuplication = (newItem) => {
+    let itemExists = false;
+
+    items.forEach((itemObject) => {
+      //Remove punctuation of existing item with regex
+      let existingItem = itemObject.itemName;
+      let cleanExistingItem = existingItem.replace(/[\W|_]/g, '');
```

Comment:
Since this regular expression is used twice `/[\W|_]/g` it can go in a variable that also describes what it does (super useful when other people check the code or the original coder checks the code after a while :smiley: )

Reply:
That's a great idea